components/chainhook-event-observer/src/chainhooks/types.rs

```diff
@@ -24,6 +25,30 @@ impl HookFormation {
         }
     }
 
+    pub fn serialized_stacks_predicates(
+        &self,
+    ) -> Vec<(&String, &StacksNetwork, &StacksTransactionFilterPredicate)> {
+        let mut stacks = vec![];
+        for chainhook in self.stacks_chainhooks.iter() {
+            stacks.push((
+                &chainhook.uuid,
+                &chainhook.network,
+                &chainhook.transaction_predicate,
+            ));
+        }
+        stacks
+    }
+
+    pub fn serialized_bitcoin_predicates(
```

Comment:
Since `serialized_stacks_predicates()` and `serialized_bitcoin_predicates()` are only called once and at the same time it may be simpler to have only one method `serialized_predicates` 

```rust
    pub fn serialize_predicates(
        &self,
    ) -> Vec<(&String, &StacksNetwork, &StacksTransactionFilterPredicate)> {
        let mut predicates = vec![];
        for chainhook in self.stacks_chainhooks.iter() {
            stacks.push((
               "stacks"
                &chainhook.uuid,
                &chainhook.network,
                &chainhook.transaction_predicate,
            ));
        }
        for chainhook in self.bitcoin_chainhooks.iter() {
            stacks.push((
               "bitcoin"
                &chainhook.uuid,
                &chainhook.network,
                &chainhook.transaction_predicate,
            ));
        }
        predicates
    }
```
   
and in handle_get_hooks.rs:

```rs
let mut predicates =  hooks
    .serialize_predicates()
    .iter()
    .map(|(chain, uuid, network, predicate)| {
        json!({
            "chain": chain,
            "uuid": uuid,
            "network": network,
            "predicate": predicate,
        })
    })
    .collect::<Vec<_>>();
```
   
 Unless it need to be decoupled at some point

Reply:
Decoupling is going to be required at some point, yes. Also, the types are a bit different, the proposal above would not pass the type check.